### DISCUSSION TITLE
Load runtimeConfig from remote api before server started

### DISCUSSION THREAD

**Discussion by tiborpino** [nuxt/nuxt#17524] [Questions]
We would like to load all runtimeConfig values from remote api before sever start.

I know about nuxt.config hooks, but I don't know how to set runtimeConfig in hooks?

Thank you for your help

Possible scenario
1. yarn start
2. nuxt.hook called
3. fetch data from remote api
4. set runtimeConfig values from fetched data
5. start server with runtimeConfig values from fetched data

**Comment by tiborpino** [comment]
Please is this possible?

**Comment by TheAlexLichter** [comment]
You could try using a custom nuxt module. The module `setup` function can be `async` and in there you could handle the API call and eventually set the respective `runtimeConfig` values ☺️ 

[Full example](https://stackblitz.com/edit/nuxt-discussion-5061?file=module.ts,nuxt.config.ts,app.vue)

**Reply by tiborpino** [reply]
The link you provided is broken.

**Reply by TheAlexLichter** [reply]
> The link you provided is broken.

Works fine for me 🤔 

https://stackblitz.com/edit/nuxt-discussion-5061

![grafik](https://user-images.githubusercontent.com/640208/170937911-ca590c6a-4ab2-4256-ae7e-99b65325c5e3.png)
